tidy3d/components/base.py

```diff
@@ -86,6 +86,28 @@ def _get_valid_extension(fname: str) -> str:
     )
 
 
+def check_previous_fields_validation(required_fields):
+    """Decorate ``validator`` to check that other fields have passed validation."""
+
+    def actual_decorator(validator):
+        @wraps(validator)
+        def _validator(cls, val, values):
+            """New validator function."""
+            for field in required_fields:
+                if field not in values:
```

Comment:
just to confirm, if `field` fails validation, it's just not put in `values`? (as opposed to put into values with `None`?)

Reply:
yeah, just double checked with pydantic documentation https://docs.pydantic.dev/1.10/usage/validators/:
> If validation fails on another field (or that field is missing) it will not be included in values, hence if 'password1' in values and ... in this example.